pkg/kubelet/dockertools/manager.go

```diff
@@ -780,7 +780,16 @@ type dockerVersion struct {
 	*semver.Version
 }
 
+// newDockerVersion returns a semantically versioned docker version value
 func newDockerVersion(version string) (dockerVersion, error) {
+	// Older versions of Docker could return non-semantically versioned values (distros like Fedora
+	// included partial values such as 1.8.1.fc21 which is not semver). Force those values to be semantic.
+	if strings.Count(version, ".") > 2 {
+		switch parts := strings.Split(version, "."); {
+		case len(parts) >= 3:
+			version = strings.Join(parts[:3], ".") + "-" + strings.Join(parts[3:], ".")
```

Comment:
It seems like that It would change prerelease versions like1.0.0-beta.2 to 1.0.0-beta-2 and might break them, although haven't tested it.